packages/overlay/src/OverlayStack.ts

```diff
@@ -180,6 +180,17 @@ class OverlayStack {
             );
             overlay.dispatchEvent(queryPathEvent);
         } else if (overlay.type === 'hint') {
+            const previous = this.stack.find((overlayEl) => {
+                return (
+                    overlayEl.type !== 'manual' &&
+                    overlayEl.triggerElement &&
+                    overlayEl.triggerElement === overlay.triggerElement
+                );
+            });
+            if (previous) {
+                overlay.open = false;
+                return;
+            }
```

Comment:
pure nit, just makes it clear that the reference to the `previous` overlay isn't important.

```suggestion
            const hasPrevious = this.stack.some((overlayEl) => {
                return (
                    overlayEl.type !== 'manual' &&
                    overlayEl.triggerElement &&
                    overlayEl.triggerElement === overlay.triggerElement
                );
            });
            if (hasPrevious) {
                overlay.open = false;
                return;
            }
```